packages/ckeditor5-ui/src/search/searchview.ts

```diff
@@ -0,0 +1,336 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/search/searchview
+*/
+
+import { FocusTracker, KeystrokeHandler, type Locale } from '@ckeditor/ckeditor5-utils';
+import View from '../view';
+import SearchFieldView from './searchfieldview';
+import SearchInfoView from './searchinfoview';
+import SearchResultsView from './searchresultsview';
+import type LabeledFieldView from '../labeledfield/labeledfieldview';
+import type InputView from '../input/inputview';
+import type FilteredView from './filteredview';
+import { escapeRegExp } from 'lodash-es';
+import FocusCycler from '../focuscycler';
+
+import '../../theme/components/search/search.css';
+
+/**
+ * A search component that allows filtering of an arbitrary view based on a search query
+ * specified by the user.
+ *
+ *```ts
+ * // This view must specify the `filter()` and `focus()` methods.
+ * const filteredView = ...;
+ *
+ * const searchView = new SearchView( locale, {
+ * 	searchFieldLabel: 'Search list items',
+ * 	filteredView
+ * } );
+ *
+ * view.render();
+ *
+ * document.body.append( view.element );
+ * ```
+ *
+ * @extends module:ui/view~View
+ */
+export default class SearchView extends View {
```

Comment:
I'm wondering if this should go a little bit further the generic path. 

## ATM

The component couples `LabeledFieldView<Any input type>` with any filterable view (view with `filter()` method`). For instance, this way you can type text and filter a list, or a toolbar, or virtually anything.

ATM you can configure `<Any input type>` and customize it a little bit, for instance, you can use numbers for the filter (`InputNumberView`).

## What this could become

This could be a base class that couples `(any)View<that communicates when the search occurs>` with any filterable view (view with `filter()` method`).

This way you could for instance use a list of clickable buttons (e.g. tags) to filter a list of some items (products, templates, whatever). There are no limits to the creativity at this point, you can use any view to filter any other view.

Then, `TextSearchView` would extend the now-generic `SearchView` and bring `LabeledFieldView<Any input type>` and do exactly the same thing the implementation in this PR would do.

This will spare us breaking changes down the road because if we want more flexibility we'll have to make `SearchView` generic and extract `TextSearchView` anyway.

Reply:
For now, let's rename it. `TextSearchView` sounds great.